test/integration/http_integration.cc

```diff
@@ -635,6 +635,121 @@ void HttpIntegrationTest::testRetry() {
   EXPECT_EQ(512U, response->body().size());
 }
 
+IntegrationStreamDecoderPtr HttpIntegrationTest::setupPerStreamIdleTimeoutTest() {
+  config_helper_.addConfigModifier(
+      [&](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager& hcm)
+          -> void {
+        auto* route_config = hcm.mutable_route_config();
+        auto* virtual_host = route_config->mutable_virtual_hosts(0);
+        auto* route = virtual_host->mutable_routes(0)->mutable_route();
+        route->mutable_idle_timeout()->set_seconds(1);
```

Comment:
You may want shorter idle timeouts but a scaleTimeoutForTsan/Asan function.   Your call.